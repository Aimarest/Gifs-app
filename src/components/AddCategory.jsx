import { useState } from "react";

 export const AddCategory = ( { handleAddCategory } ) => {

    const [newCategory, setNewCategory] = useState('');

    const handleAdd = (event) => {

        setNewCategory(event.target.value);
      
    }
    const onSubmit = (event) => {
        event.preventDefault(); 
       
        if( newCategory.trim().length < 1 ) return;   //trim() lo que hace es borrar los espacios en blanco que haya.
        handleAddCategory( newCategory.trim() )
        setNewCategory('');
    }
    const handleClick = () =>{
        if( newCategory.trim().length < 1 ) return;   
        handleAddCategory( newCategory.trim() )
        setNewCategory('');
    }
    return (
        <form onSubmit={onSubmit}>
            <input type='text' placeholder='Buscar gifs' onChange={handleAdd} value={newCategory}></input>
            <button onClick={ handleClick }> Add a new category </button>
        </form>
    )
}
