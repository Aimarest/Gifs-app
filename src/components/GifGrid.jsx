
import { useEffect, useState } from "react";
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem }from "./gifItem";


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category )


    return (
        <>
            <h3>{ category }</h3>

            {
                isLoading && ( <h2>Cargando im√°genes...</h2> )  //Es una forma de hacer un if o un operador ternario corto.
            }
            <ul className="card-grid">
                {
                    images.map( ( image ) => (
                    <GifItem 
                    key={ image.id } 
                    {... image} //ESTO SE UTILIZA PARA ESPARCIR LAS PROPIEDADES DE IMAGE EN EL COMPONENTE AL QUE LO ESTAMOS PASANDO.
                    /> 
                    ))
                }
            </ul>
        </>
    );
}

