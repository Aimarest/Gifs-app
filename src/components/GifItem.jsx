
import PropTypes  from 'prop-types';
export const GifItem = ({ title, url, id })  => {

return (
     <div className="card">
       
        <h3>{title}</h3>
        <img alt={title} src={url} />
    </div>
 
)}
   
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}




