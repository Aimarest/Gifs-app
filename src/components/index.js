//Esto es un archivo de barril para poder exportar todo a la vez.

export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';