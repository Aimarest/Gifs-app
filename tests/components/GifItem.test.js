import { GifItem } from "../../src/components/GifItem";
import {  render } from '@testing-library/react';

describe('Pruebas sobre componente GifItem', () => {
    const title = 'joey';
    const url = 'https://media0.giphy.com/media/W3a0zO282fuBpsqqyD/giphy.gif?cid=9dbd306dzzcjjlswlve9yss3bztxhwdhsuk059upzh91e2y1&rid=giphy.gif&ct=g'
    test('Debe hacer match con el snapshot', () => {

        const { container } = render(

            <GifItem title={ title } url={ url }/>
        )
        expect(container).toMatchSnapshot();
    })
});
