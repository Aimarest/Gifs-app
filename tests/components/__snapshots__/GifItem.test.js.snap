// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h3>
      joey
    </h3>
    <img
      alt="joey"
      src="https://media0.giphy.com/media/W3a0zO282fuBpsqqyD/giphy.gif?cid=9dbd306dzzcjjlswlve9yss3bztxhwdhsuk059upzh91e2y1&rid=giphy.gif&ct=g"
    />
  </div>
</div>
`;
